File: backend/src/main/java/com/prepkeeper/entity/Reminder.java
package com.prepkeeper.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

/**
 * 提醒记录实体
 */
@Data
@Entity
@Table(name = "reminders")
public class Reminder {

    @Id
    @Column(name = "reminder_id", length = 64)
    private String reminderId;

    @Column(name = "user_id", length = 64, nullable = false)
    private String userId;

    @Column(name = "plan_id", length = 64, nullable = false)
    private String planId;

    /**
     * 提醒类型: 1-每日提醒 2-催更提醒 3-周报提醒
     */
    @Column(name = "reminder_type", nullable = false)
    private Integer reminderType;

    @Column(name = "trigger_time", nullable = false)
    private LocalDateTime triggerTime;

    @Column(name = "content", length = 1000)
    private String content;

    /**
     * 是否已发送: 0-否 1-是
     */
    @Column(name = "is_sent")
    private Integer isSent = 0;

    @Column(name = "sent_at")
    private LocalDateTime sentAt;
}

File: agent-service/app/models/reminder.py
"""
提醒记录模型
"""
from sqlalchemy import Column, String, Integer, TIMESTAMP, SmallInteger, ForeignKey, Text
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.models.database import Base
import uuid


class Reminder(Base):
    """提醒记录表"""
    __tablename__ = "reminders"

    reminder_id = Column(String(64), primary_key=True, default=lambda: str(uuid.uuid4()))
    user_id = Column(String(64), ForeignKey("users.user_id"), nullable=False)
    plan_id = Column(String(64), ForeignKey("study_plans.plan_id"), nullable=False)
    reminder_type = Column(SmallInteger, nullable=False)  # 1-每日提醒 2-催更提醒 3-周报提醒
    trigger_time = Column(TIMESTAMP, nullable=False)
    content = Column(String(1000), nullable=True)
    is_sent = Column(SmallInteger, default=0)  # 0-否 1-是
    sent_at = Column(TIMESTAMP, nullable=True)

    # 关系
    user = relationship("User", backref="reminders")
    plan = relationship("StudyPlan", backref="reminders")

    def __repr__(self):
        return f"<Reminder(reminder_id={self.reminder_id}, type={self.reminder_type})>"